modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Enrich.scala

```diff
@@ -64,33 +68,40 @@ object Enrich {
    * [[Assets.State.make]] downloads assets for the first time unconditionally during
    * [[Environment]] initialisation, then if `assetsUpdatePeriod` has been specified -
    * they'll be refreshed periodically by [[Assets.updateStream]]
-   *
-   * @param ordered indicates whether the events should be processed ordered or not
    */
-  def run[F[_]: Concurrent: ContextShift: Clock: Parallel, A](env: Environment[F, A], ordered: Boolean): Stream[F, Unit] = {
+  def run[F[_]: Concurrent: ContextShift: Clock: Parallel: Timer, A](env: Environment[F, A]): Stream[F, Unit] = {
     val registry: F[EnrichmentRegistry[F]] = env.enrichments.get.map(_.registry)
     val enrich: Enrich[F] = {
       implicit val rl: RegistryLookup[F] = env.registryLookup
       enrichWith[F](registry, env.igluClient, env.sentry, env.processor)
     }
 
-    val enrichPipe: Pipe[F, (A, Array[Byte]), (A, Result)] =
-      in => {
-        val parallel =
-          if (ordered)
-            in.parEvalMap[F, (A, Result)](env.streamsSettings.concurrency.enrichment) _
-          else
-            in.parEvalMapUnordered[F, (A, Result)](env.streamsSettings.concurrency.enrichment) _
-        parallel { case (orig, bytes) => enrich(bytes).map((orig, _)) }
-      }
-
-    env.source
-      .pauseWhen(env.pauseEnrich)
-      .evalTap(_ => env.metrics.rawCount)
-      .fproduct(env.getPayload)
-      .through(enrichPipe)
-      .through(sinkResult(sinkOne(env), env.metrics.enrichLatency, env.streamsSettings.concurrency.output))
-      .through(env.checkpointer)
+    val enriched =
+      env.source.chunks
+        .prefetch
+        .evalTap(chunk => Logger[F].debug(s"${DateTime.now().toString()} - Starting to process chunk of size ${chunk.size}"))
```

Comment:
I added the timestamp because it was not appearing on my local machine. It might get added by Cloudwatch and Stackdriver in the UI. Anyway I'm removing it